### DISCUSSION TITLE
Refactoring and clarifying labels

### DISCUSSION THREAD

**Discussion by Falke-Design** [Leaflet/Leaflet#8018] [General]
(Spun off from https://github.com/Leaflet/Leaflet/issues/7830)

We use this discussion for refactoring and "discussing" about labels and their usage. We want to improve with this the workflow.

@jonkoops @Malvoz @mourner @IvanSanchez @johnd0e feel free to contribute.

**Comment by Falke-Design** [comment]
## https://github.com/Leaflet/Leaflet/labels/in%20progress
Should we replace "in progress" with the **Draft** state?

**Reply by mourner** [reply]
Yeah, I don't think it serves much purpose now. It was introduced back when GitHub had no drafts or even linked PR/issue pairs.

**Reply by Malvoz** [reply]
Thanks @mourner. Hope you are well, given the circumstances. 🙏🏻

- [x] I think we should convert these PRs to drafts, and then delete this label.
      Edit: Done!

**Comment by Falke-Design** [comment]
## https://github.com/Leaflet/Leaflet/labels/blocker
What is the correct usage?

**Reply by mourner** [reply]
Blocker is a critical issue that needs to be resolved ASAP, and no new release can come out until it is solved. E.g. broken interaction in a major browser that makes the map unusable.

**Reply by Malvoz** [reply]
We should write a description for this label, based on the above description.

**Reply by Falke-Design** [reply]
`Critical - Needs to be resolved before releasing the next release`. I removed "issue" because it can be a PR too

**Reply by Malvoz** [reply]
Maybe "Critical issue or PR that must be resolved before the next release"?  Since labels are also applicable to discussions, I don't think there are critical discussions in the sense of GH discussions?

**Reply by Malvoz** [reply]
- [x] Added `Critical issue or PR that must be resolved before the next release` description to the `blocker` label per above comment, feel free to discuss if you think it needs changing.
- [x] With the assumption that issues/PRs with this label must be checked manually - I suggest that we add a check in [RELEASE.md](https://github.com/Leaflet/Leaflet/blob/main/RELEASE.md) to ensure these issues/PRs are never skipped on releases.
    Edit: fixed in https://github.com/Leaflet/Leaflet/pull/8019.

**Reply by Malvoz** [reply]
I just want to note that this label should generally only be used for issues and PRs (this is also stated in the description of the label itself), and not discussions.

Discussion https://github.com/Leaflet/Leaflet/discussions/8044 has this label, however if you follow the link (https://github.com/Leaflet/Leaflet/labels/blocker) from RELEASE.md, you wont see that there's a _discussion_ with a blocker label.

Discussions are great, but if there are blockers there should be corresponding issues/PRs.

**Comment by Falke-Design** [comment]
## https://github.com/Leaflet/Leaflet/labels/accessibility
Should be below the accessibility label only accessibility things or should we maybe inlcude HTML / CSS validation errors? Do we have a better name which inlcudes both, do we want to include both?

**Reply by mourner** [reply]
I think this should be generally about accessibility improvements, tangential (but sometimes overlapping) to validation errors.

**Reply by Malvoz** [reply]
- [x] Added description: `Anything related to ensuring no barriers exist that prevent interactions or information access`. Feedback welcome.

**Comment by Falke-Design** [comment]
## https://github.com/Leaflet/Leaflet/labels/needs%20review
Definition?

**Reply by mourner** [reply]
Also an outdated label. Now we have PR reviews.

**Reply by Malvoz** [reply]
Ok, let's delete this label.

Edit: done!

**Reply by Falke-Design** [reply]
Maybe we can now use this label for `ready for merge` / `quick review`

**Reply by Falke-Design** [reply]
Ok done, I thought if we reuse the label, we can discuss it here 😉

**Comment by Malvoz** [comment]
## https://github.com/Leaflet/Leaflet/labels/compatibility

Is this about cross-browser compatibility? Backwards compatibility? Device compatibility? Something else?

**Reply by mourner** [reply]
Cross-browser/device compatibility as far as I remember. Maybe worth renaming to `environment-specific`.

**Reply by Malvoz** [reply]
I think `compatibility` is fine, as long as we provide the label a description. Although I don't oppose to changing it to `environment-specific`.

**Reply by Malvoz** [reply]
- [x] Added `Cross-browser/device/environment compatibility` description to the `compatibility` label for now. Feedback welcome.

**Comment by Falke-Design** [comment]
## https://github.com/Leaflet/Leaflet/labels/ux
Definition?

**Reply by Malvoz** [reply]
General user experience issues?

I think it's closely related to the accessibility label, however things can be user experience issues without being accessibility issues.
For example dark mode (https://github.com/Leaflet/Leaflet/issues/7577) I think is more a UX issue than an accessibility issue. Or something that looks "ugly" or has unexpected behavior is a UX issue (although the latter can sometimes result in accessibility issues).

**Comment by Falke-Design** [comment]
## `Ready for merge` / `quick review` label

There should be a label which can be used to mark some PR which can be merged without much effort.

**Comment by Malvoz** [comment]
## https://github.com/Leaflet/Leaflet/labels/later

This label isn't used widely. And it's not clear to me when "later" is or what that really means. Perhaps this label was added before milestones were a thing?

Maybe it's useful and should be kept, but no harm in discussing.

Edit: Can be useful in conjunction to noting that an issue/PR is blocked on another issue/PR.
(https://github.com/orgs/github-community/discussions/4928)

**Comment by Malvoz** [comment]
### `invalid` label

A new label to close out issues that are invalid, e.g. issues that are out of scope for Leaflet.

We could use [GitHub Actions to automatically comment on issues when a specific label is applied](https://docs.github.com/en/actions/managing-issues-and-pull-requests/commenting-on-an-issue-when-a-label-is-added).

There can be many reasons why an issue could be invalid.

<details>
<summary>Example comments from Iván</summary><br>

> Hi, great to hear that you find Leaflet useful!
> 
> However, this issue tracker is used for reporting bugs and discussing new features. For questions on using Leaflet, please use [gis.stackexchange.com](https://gis.stackexchange.com/questions/tagged/leaflet) or [stackoverflow](https://stackoverflow.com/questions/tagged/leaflet).
> 
> If you are *really* sure that this is a bug in leaflet, or a feature request, please [change the wording to make it look like a bug report](http://www.chiark.greenend.org.uk/~sgtatham/bugs.html).

_Originally posted by @IvanSanchez in https://github.com/Leaflet/Leaflet/issues/8139#issuecomment-1098326178_

> Hi, and thanks for taking the time to open a bug report in Leaflet.
> 
> However, in this repository we only handle bugs in "[vanilla](http://vanilla-js.com/)" Leaflet. This means that we do **not** handle bugs which are specific to frameworks such as:
> * React
> * Ractive
> * AngularJS
> * Bootstrap
> * Leaflet for R
> * Joomla
> * Wordpress
> * Ionic
> * Cordova
> * Vue
> * Electron
> * Polymer
> * TypeScript
> * D3
> * Folium
> * NativeScript
> * ...etc
> 
> Please understand that we only have the time and energy to test Leaflet in plain web browsers.
> 
> Please try to either reproduce the bug without using any frameworks, or submit a bug to the appropriate repo.

_Originally posted by @IvanSanchez in https://github.com/Leaflet/Leaflet/issues/8077#issuecomment-1073552630_

</details>

The comment from the `invalid` label could have multiple sections explaining the reasons for why it is invalid, if necessary.

**Reply by Falke-Design** [reply]
Good idea! But maybe we find a more positiv word then `invalid` like `off-topic` or so

**Reply by Malvoz** [reply]
@Falke-Design I suggested `invalid` because that's the name I've seen in other repos. 😋

**Reply by Malvoz** [reply]
Also, `invalid` covers issues such as https://github.com/Leaflet/Leaflet/issues/7562 where no useful information has been provided, whereas `off-topic` wouldn't necessarily make sense.

Edit: `invalid` is actually one of [Github's default labels](https://docs.github.com/en/issues/using-labels-and-milestones-to-track-work/managing-labels#about-default-labels).

**Comment by Malvoz** [comment]
## dependabot labels <a href="https://github.com/dependabot"><img width="25" src="https://user-images.githubusercontent.com/26493779/164191187-8c71f6ae-379d-447f-b4a4-ae075e3eea91.png"></a>

> Looks like good guy dependabot keeps adding new labels: https://github.com/Leaflet/Leaflet/labels/javascript, https://github.com/Leaflet/Leaflet/labels/github_actions? Is this necessary? Is this desired? Can it be disabled otherwise?

_Originally posted by @Malvoz in https://github.com/Leaflet/Leaflet/issues/8132#issuecomment-1094988627_

> @Malvoz Yeah we can disable this by setting the [labels option](https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file#labels).

_Originally posted by @jonkoops in https://github.com/Leaflet/Leaflet/issues/8132#issuecomment-1096927067_

**Reply by jonkoops** [reply]
I think we should remove these labels, see #8199

**Reply by jonkoops** [reply]
These labels have been removed, marking this conversation as resolved.
